### PR TITLE
add terminateProcessId

### PR DESCRIPTION
**Pull request by joeyh**
closes https://github.com/haskell/win32/issues/91

### CODE REVIEW

**Comment by joeyh** [comment]
This builds, but linking against it is failing, I'll try to fix thaat

File: System/Win32/Process.hsc [hunk 1]
```diff
@@ -81,6 +81,15 @@ getCurrentProcessId = c_GetCurrentProcessId
 getCurrentProcess :: IO ProcessHandle
 getCurrentProcess = c_GetCurrentProcess
 
+foreign import WINDOWS_CCONV unsafe "windows.h TerminateProcess"
+    c_TerminateProcess :: ProcessHandle -> Int -> IO Int
+
+terminateProcessId :: ProcessId -> IO ()
```
Comment (Mistuke) [inline System/Win32/Process.hsc]: call it `terminateProcessById`.

File: System/Win32/Process.hsc [hunk 2]
```diff
@@ -81,6 +81,15 @@ getCurrentProcessId = c_GetCurrentProcessId
...
+terminateProcessId p = bracket
+    (openProcess pROCESS_TERMINATE False p)
+    closeHandle
+    (\h -> void $ c_TerminateProcess h 1)
```
Comment (Mistuke) [inline System/Win32/Process.hsc]: don't swallow the exception, change `void` to `failIfFalse_ "c_TerminateProcess"`

**Comment by Mistuke** [comment]
Thanks @joeyh !

Aside from these small changes can you add a changelog entry? Something like

```
## Unreleased GIT

* Add `terminateProcessById` (See #91)
```

**Comment by joeyh** [comment]
Thanks @Mistuke for a really excellent review. I've taken on board all
the suggestions.

-- 
see shy jo

**Pull request merged**